packages/remark-captions/dist/index.js

```diff
@@ -35,8 +35,8 @@ function internLegendVisitor(internalBlocks) {
     if (parent && parent.type === 'figure') return;
 
     // legend can only be in a paragraph
-    var lastP = getLast(node.children);
-    if (!lastP || lastP.type !== 'paragraph') return;
+    var lastP = node.children ? getLast(node.children) : parent;
+    if (!lastP || node.children && lastP.type !== 'paragraph' || !node.children && parent.type !== 'paragraph') return;
```

Comment:
can you put a comment as a "4 conditional assertion with use of priority" is not the easyest thing to read.